commit 7d395954136c45e1533f355068399fec5e606db1
Author: Baptiste Assmann <bedis9@gmail.com>
Date:   Fri Jun 22 13:03:50 2018 +0200

    MINOR: dns: fix wrong score computation in dns_get_ip_from_response
    
    dns_get_ip_from_response() is used to compare the caller current IP to
    the IP available in the records returned by the DNS server.
    A scoring system is in place to get the best IP address available.
    That said, in the current implementation, there are a couple of issues:
    1. a comment does not match what the code does
    2. the code does not match what the commet says (score value is not
       incremented with '2')
    
    This patch fixes both issues.
    
    Backport status: 1.8
    
    (cherry picked from commit 84221b4e9010810cf93b7ad7a31d825fa9fc26bf)
    [wt: Baptiste explicitly asked for this one to be backported to stable]
    Cc: Baptiste <bedis9@gmail.com>
    Signed-off-by: Willy Tarreau <w@1wt.eu>

diff --git a/src/dns.c b/src/dns.c
index 153a46b2..d8388ef1 100644
--- a/src/dns.c
+++ b/src/dns.c
@@ -1027,10 +1027,13 @@ int dns_get_ip_from_response(struct dns_response_packet *dns_p,
 		}
 
 		/* Check if the IP found in the record is already affected to a
-		 * member of a group.  If yes, the score should be incremented
+		 * member of a group.  If not, the score should be incremented
 		 * by 2. */
-		if (owner && snr_check_ip_callback(owner, ip, &ip_type))
+		if (owner && snr_check_ip_callback(owner, ip, &ip_type)) {
 			continue;
+		} else {
+			score += 2;
+		}
 
 		/* Check for current ip matching. */
 		if (ip_type == currentip_sin_family &&
